src/payment/strategies/adyenv2/adyenv2-payment-strategy.ts

```diff
@@ -143,66 +141,37 @@ export default class AdyenV2PaymentStrategy implements PaymentStrategy {
                 return this._store.dispatch(this._paymentActionCreator.submitPayment(paymentPayload));
             })
             .catch(error => {
-                if (!(error instanceof RequestError) || (!some(error.body.errors, { code: 'three_d_secure_required' }) && !some(error.body.errors, { code: 'additional_action_required' }))) {
+                if (!(error instanceof RequestError) || !some(error.body.errors, { code: 'additional_action_required' })) {
                     return Promise.reject(error);
                 }
 
-                if (error.body.three_ds_result.code === ResultCode.IdentifyShopper) {
-                    return this._handle3DS2Fingerprint(error.body.three_ds_result, payment.methodId)
-                        .then((payment: Payment) =>
-                            this._store.dispatch(this._paymentActionCreator.submitPayment({
-                                ...payment,
-                                paymentData: {
-                                    ...payment.paymentData,
-                                    shouldSaveInstrument,
-                                },
-                            })))
-                        .catch(error => {
-                            if (!(error instanceof RequestError) || !some(error.body.errors, { code: 'three_d_secure_required' })) {
-                                return Promise.reject(error);
-                            }
-
-                            if (error.body.three_ds_result.code === ResultCode.ChallengeShopper) {
-                                return this._handle3DS2Challenge(error.body.three_ds_result, payment.methodId)
-                                    .then((payment: Payment) =>
-                                        this._store.dispatch(this._paymentActionCreator.submitPayment({
-                                            ...payment,
-                                            paymentData: {
-                                                ...payment.paymentData,
-                                                shouldSaveInstrument,
-                                            },
-                                        }))
-                                    );
-                            }
-
+                return this._handleFromAction(error.body.provider_data)
+                    .then((payment: Payment) =>
```

Comment:
You can drop the `Payment` type declaration because it's already implied by the return value of `_handleFromAction`